apps/server/src/shared/infra/collaborative-storage/strategy/nextcloud/nextcloud.client.ts

```diff
@@ -405,7 +429,7 @@ export class NextcloudClient {
 			})
 			.catch((err) => {
 				error(err);
-				throw new NotImplementedException();
+				throw new Error();
```

Comment:
Why you did changed this?

Reply:
Nearly all our code is bind to a (user http) request. To response for that parts with 4xx, 5xx is ok. 
I am not sure if NotImplementedException is the best for this place, but i do not check the code/case.
But new Error without any information is also not helpfull for any developer, or user. This is also used in line 428 and can be improved. 
If this change as part of this ticket, this also increase the testing scope of this ticket. Let say any part that use this api point and has a error handling for this specific case, than it must also changed. I do not thinks so, but it is possible and a "fast" fix can be result in bigger trouble.

Reply:
I rolled it back because, if it leads to this much discussion, it is probably too much for an off-topic change. And also there are other usages of `NotImplementedException` in this class which are questionable. So they are better checked at a whole.

And I'm also not familiar enough with this class to say what information should be put in an error here.